rbx_reflector/src/patches.rs

```diff
@@ -0,0 +1,150 @@
+        for (class_name, class_changes) in &self.change {
+            let class = database
+                .classes
+                .get_mut(class_name.as_str())
+                .ok_or_else(|| {
+                    anyhow!(
+                        "Class {} modified in patch file does not exist in database",
+                        class_name
+                    )
+                })?;
+
+            for (property_name, property_change) in class_changes {
+                let existing_property = class
+                    .properties
+                    .get_mut(property_name.as_str())
+                    .ok_or_else(|| {
+                        anyhow!(
+                            "Property {}.{} modified in patch file does not exist in database",
+                            class_name,
+                            property_name
+                        )
+                    })?;
+
+                if let Some(data_type) = &property_change.data_type {
+                    existing_property.data_type = data_type.clone();
+                }
+
+                if let Some(kind) = property_change.kind() {
+                    if let (
+                        PropertyKind::Canonical { serialization },
+                        PropertyKind::Canonical {
+                            serialization: existing_serialization,
+                        },
+                    ) = (&kind, &existing_property.kind)
+                    {
+                        match (serialization, existing_serialization) {
+                            (PropertySerialization::Serializes, PropertySerialization::Serializes)
+                            | (PropertySerialization::DoesNotSerialize, PropertySerialization::DoesNotSerialize) => bail!("The serialization for property {class_name}.{property_name} was unchanged"),
+                            _ => {}
+                        };
+                    }
+
+                    existing_property.kind = kind;
+                }
+
+                if let Some(scriptability) = &property_change.scriptability {
+                    match (existing_property.scriptability, scriptability) {
+                        (Scriptability::Custom, Scriptability::Custom)
+                        | (Scriptability::None, Scriptability::None)
+                        | (Scriptability::Read, Scriptability::Read)
+                        | (Scriptability::ReadWrite, Scriptability::ReadWrite)
+                        | (Scriptability::Write, Scriptability::Write) => bail!("The scriptability for property {class_name}.{property_name} was unchanged"),
+                        _ => {}
+                    };
+
+                    existing_property.scriptability = *scriptability;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "PascalCase", deny_unknown_fields)]
+struct Patch {
+    #[serde(default)]
```

Comment:
I'm curious about the decision to move to only supporting `Change` patches and not `Add`. I'm guessing that it's because the full API dump renders it mostly moot?

Reply:
Yes, I did since all properties exist in the new full dump.